backend/services/user.py

```diff
@@ -49,6 +50,24 @@ def get(self, pid: int) -> UserDetails | None:
             user_details = UserDetails(**user_fields)
             return user_details
 
+    def get_by_id(self, id: int) -> User:
+        """Get a User by their id.
+
+        Args:
+            id: The ID of the user.
+
+        Returns:
+            User
+
+        Raises:
+            ResourceNotFoundException if the User ID is not found
+        """
+        user_entity = self._session.get(UserEntity, id)
```

Comment:
Do we need permissions for this particular call? Such as if there is sensitive data here

Reply:
Hmm interesting question. In the usage, we're not leaking anything, but this point is well taken.

I think `UserService` is due for a cleanup and refactor (such as taking up the idiom of using `subject` for most methods).

The use case we care about here is an organizer or admin being able to add an attendee. For this reason, and some other scenarios (such as searching for friends), I believe we'll want registered users to be able to lookup other users (but perhaps only get a subset of their profile data).

This is something we should come back to and tidy up after the semester.